### PR TITLE
Add aliases for Install-PSResource, Find-PSResource, Update-PSResource, Publish-PSResource

### LINKED ISSUE #1152: Support 'inmo', 'fimo', 'upmo', 'pumo'
### Summary of the new feature / enhancement

Consider support 'inmo', 'fimo', 'upmo', 'pumo'

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Add 'inps', 'fips', 'upps' and 'pups' aliases for Install-PSResource, Find-PSResource, Update-PSResource, Publish-PSResource respectively.

## PR Context
Resolves #1152 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Comment by ThomasNieto** [comment]
I don't think `ps` is the best alias postfix as `ps` is short for PowerShell not resources. It might be confusing for users thinking it may update PowerShell. Have you thought of using something like `rs` or `res`?

**Review by ThomasNieto — CHANGES_REQUESTED** [review]
You'll also need to add the aliases to the module manifest.

**Comment by ThomasNieto** [comment]
The alias prefix also doesn't match the [verb alias recommendations](https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.3).

> Each approved verb has a corresponding alias prefix defined. We use this alias prefix in aliases for commands using that verb. For example, the alias prefix for Import is ip and, accordingly, the alias for Import-Module is ipmo. This is a recommendation but not a rule; in particular, it need not be respected for command aliases mimicking well known commands from other environments.

```powershell
Get-Verb Install, Find, Update, Publish | select Verb, AliasPrefix

Verb    AliasPrefix
----    -----------
Find    fd
Publish pb
Update  ud
Install is
```

**Review by anamnavi — APPROVED** [review]

File: src/code/FindPSResource.cs [hunk 1]
```diff
@@ -21,6 +21,7 @@ namespace Microsoft.PowerShell.PSResourceGet.Cmdlets
     [Cmdlet(VerbsCommon.Find,
         "PSResource",
         DefaultParameterSetName = NameParameterSet)]
+    [Alias("fips")]
```
Comment (anamnavi) [inline src/code/FindPSResource.cs]:
```suggestion
    [Alias("fdps")]
```

File: src/code/InstallPSResource.cs [hunk 2]
```diff
@@ -18,7 +18,11 @@ namespace Microsoft.PowerShell.PSResourceGet.Cmdlets
     /// The Install-PSResource cmdlet installs a resource.
     /// It returns nothing.
     /// </summary>
-    [Cmdlet(VerbsLifecycle.Install, "PSResource", DefaultParameterSetName = "NameParameterSet", SupportsShouldProcess = true)]
+    [Cmdlet(VerbsLifecycle.Install, 
+        "PSResource", 
+        DefaultParameterSetName = "NameParameterSet", 
+        SupportsShouldProcess = true)]
+    [Alias("inps")]
```
Comment (anamnavi) [inline src/code/InstallPSResource.cs]:
```suggestion
    [Alias("isps")]
```

File: src/code/PublishPSResource.cs [hunk 3]
```diff
@@ -23,7 +23,10 @@ namespace Microsoft.PowerShell.PSResourceGet.Cmdlets
     /// <summary>
     /// Publishes a module, script, or nupkg to a designated repository.
     /// </summary>
-    [Cmdlet(VerbsData.Publish, "PSResource", SupportsShouldProcess = true)]
+    [Cmdlet(VerbsData.Publish, 
+        "PSResource", 
+        SupportsShouldProcess = true)]
+    [Alias("pups")]
```
Comment (anamnavi) [inline src/code/PublishPSResource.cs]:
```suggestion
    [Alias("pbps")]

```

File: src/code/UpdatePSResource.cs [hunk 4]
```diff
@@ -20,6 +20,7 @@ namespace Microsoft.PowerShell.PSResourceGet.Cmdlets
     [Cmdlet(VerbsData.Update,
         "PSResource",
         SupportsShouldProcess = true)]
+    [Alias("upps")]
```
Comment (anamnavi) [inline src/code/UpdatePSResource.cs]:
```suggestion
    [Alias("udps")]

```

File: test/FindPSResourceTests/FindPSResourceLocal.Tests.ps1 [hunk 5]
```diff
@@ -234,4 +234,8 @@ Describe 'Test Find-PSResource for local repositories' -tags 'CI' {
             $item.ParentResource.Includes.DscResource | Should -Contain $dscResourceName
         }
     }
+
+    It "Get definition for alias 'fips'" {
+        (Get-Alias fips).Definition | Should -BeExactly 'Find-PSResource'
```
Comment (anamnavi) [inline test/FindPSResourceTests/FindPSResourceLocal.Tests.ps1]:
```suggestion
    It "Get definition for alias 'fdps'" {
        (Get-Alias fdps).Definition | Should -BeExactly 'Find-PSResource'
```

File: test/InstallPSResourceTests/InstallPSResourceLocal.Tests.ps1 [hunk 6]
```diff
@@ -263,4 +263,8 @@ Describe 'Test Install-PSResource for local repositories' -tags 'CI' {
         $res.Name | Should -Contain $testModuleName
         $res.Version | Should -Be "1.0.0"
     }
+
+    It "Get definition for alias 'inps'" {
+        (Get-Alias inps).Definition | Should -BeExactly 'Install-PSResource'
```
Comment (anamnavi) [inline test/InstallPSResourceTests/InstallPSResourceLocal.Tests.ps1]:
```suggestion
    It "Get definition for alias 'isps'" {
        (Get-Alias isps).Definition | Should -BeExactly 'Install-PSResource'
```

File: test/PublishPSResourceTests/PublishPSResource.Tests.ps1 [hunk 7]
```diff
@@ -615,4 +615,8 @@ Describe "Test Publish-PSResource" -tags 'CI' {
 
         {Publish-PSResource -Path $psm1Path -Repository $testRepository2 -ErrorAction Stop} | Should -Throw -ErrorId "InvalidPublishPath,Microsoft.PowerShell.PSResourceGet.Cmdlets.PublishPSResource"
     }
+
+    It "Get definition for alias 'pups'" {
+        (Get-Alias pups).Definition | Should -BeExactly 'Publish-PSResource'
```
Comment (anamnavi) [inline test/PublishPSResourceTests/PublishPSResource.Tests.ps1]:
```suggestion
    It "Get definition for alias 'pbps'" {
        (Get-Alias pbps).Definition | Should -BeExactly 'Publish-PSResource'
```

File: test/UpdatePSResourceTests/UpdatePSResourceLocalTests.ps1 [hunk 8]
```diff
@@ -290,4 +290,8 @@ Describe 'Test Update-PSResource for local repositories' -tags 'CI' {
         $res.Name | Should -Contain $moduleName
         $res.Version | Should -Be "5.0.0"
     }
+
+    It "Get definition for alias 'upps'" {
+        (Get-Alias upps).Definition | Should -BeExactly 'Update-PSResource'
```
Comment (anamnavi) [inline test/UpdatePSResourceTests/UpdatePSResourceLocalTests.ps1]:
```suggestion
    It "Get definition for alias 'udps'" {
        (Get-Alias udps).Definition | Should -BeExactly 'Update-PSResource'
```

**Review by anamnavi — CHANGES_REQUESTED** [review]

**Comment by anamnavi** [comment]
Thanks for the feedback @ThomasNieto . I've suggested changes to update the aliases to match the PowerShell recommended verbs aliases.

I also like the idea of not having "ps" denote "PSResource"...I agree it's not what my mind goes to first, I think of "powershell". Maybe "res"?

**Comment by alerickson** [comment]
@ThomasNieto we'll chat about it with @SydneyhSmith, I was just copying the format of first two letters of the verb + first two letters of the noun,  but given that doc if we're changing the first part we may as well make the second half a bit more intuitive too

File: src/code/FindPSResource.cs (hunk 1 above)
Comment (alerickson) [reply src/code/FindPSResource.cs]: added in a new commit

**Review by ThomasNieto — CHANGES_REQUESTED** [review]
The aliases look good to me, the last thing needed is to update the module manifest with the new aliases.

https://github.com/PowerShell/PSResourceGet/blob/5c783542f6d414b7146a405dcb867e98caac27ea/src/Microsoft.PowerShell.PSResourceGet.psd1#L39

**Comment by anamnavi** [comment]
@alerickson can we add the aliases here too: https://github.com/alerickson/PSResourceGet/blob/addAliases/src/Microsoft.PowerShell.PSResourceGet.psd1#L40

**Review by anamnavi — APPROVED** [review]
LGTM

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**